### DISCUSSION TITLE
EvaluateScriptAsync returns null

### DISCUSSION THREAD

**Discussion by TonyValenti** [cefsharp/CefSharp#3508] [Q&A]
I'm using EvaluateScriptAsync on this code:
```
var documenttable = document.evaluate("//*[@id='document-table']", document, null, XPathResult.FIRST_ORDERED_NODE_TYPE).singleNodeValue;

var scope = angular.element(documenttable).scope();

var data = scope.$data;

angular.toJson(data)
```

To extract an angular scope in C# but I'm getting a "null" result.  What is the right way to return data from EvaluateScriptAsync?

I'm using CEFSharp 79.1.

**Comment by amaitland** [comment]
See https://github.com/cefsharp/CefSharp/wiki/General-Usage#2-how-do-you-call-a-javascript-method-that-returns-a-result for examples.

Think of it like executing code in DevTools, you don't need a return statement. For multi line scripts I recommend using an IIFE.

https://developer.mozilla.org/en-US/docs/Glossary/IIFE
